### PR TITLE
use multiline in regex matching

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@brenton @smarterclayton review please.  I'd like to get this fixed now, i'l revisit using the docker libs to parse later.

@brenton I think the ^ is still desired, in your version you removed it.  agree?

File: pkg/build/builder/docker.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 const urlCheckTimeout = 16 * time.Second
 
 // imageRegex is used to substitute image names in buildconfigs with immutable image ids at build time.
-var imageRegex = regexp.MustCompile(`^FROM\s+\w+.+`)
+var imageRegex = regexp.MustCompile(`(?mi)^FROM\s+\w+.+`)
```
Comment (smarterclayton) [inline pkg/build/builder/docker.go]: Only the last FROM in a file takes effect.  Does this capture that?

Comment (bparees) [inline pkg/build/builder/docker.go]: well this will replace all the FROMs i believe, so "yes"?

**Comment by bparees** [comment]
I guess if you put whitespace before your FROM this won't replace that...  updating again.

**Comment by bparees** [comment]
updated.

**Comment by mfojtik** [comment]
@kargakis is going to look into using the Docker parser to replace this. maybe add FIXME comment there.

**Comment by bparees** [comment]
@mfojtik i opened an issue and assigned him, so it's tracked.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by brenton** [comment]
Yeah, it's still useful to have the ^ in there right now.  In my local testing I had it after the (?m) and it was messing things up so I removed it from my environment while I was debugging.  I liike @mfojtik's idea about adding the FIXME.

**Pull request merged**